cli/src/main/java/com/devonfw/tools/ide/process/ProcessContextImpl.java

```diff
@@ -218,13 +219,21 @@ public ProcessResult run(ProcessMode processMode) {
    * "https://stackoverflow.com/questions/14165517/processbuilder-forwarding-stdout-and-stderr-of-started-processes-without-blocki/57483714#57483714">StackOverflow</a>
    *
    * @param is {@link InputStream}.
+   * @param errorStream to identify if the output came from stdout or stderr
    * @return {@link CompletableFuture}.
    */
-  private static CompletableFuture<List<String>> readInputStream(InputStream is) {
+  private static CompletableFuture<List<String>> readInputStream(InputStream is, boolean errorStream, ConcurrentLinkedQueue<OutputMessage> outputMessages) {
 
     return CompletableFuture.supplyAsync(() -> {
 
       try (InputStreamReader isr = new InputStreamReader(is); BufferedReader br = new BufferedReader(isr)) {
+
+        String line;
+        while ((line = br.readLine()) != null) {
+          OutputMessage outputMessage = new OutputMessage(errorStream, line);
+          outputMessages.add(outputMessage);
+        }
+
         return br.lines().toList();
```

Comment:
why do we try to read the lines again from the buffered reader after it has already been consumed to the end?
IMHO you should change the signature from `CompletableFuture<List<String>>` to `CompletableFuture<Void>` and simply return `null` here.